### PR TITLE
do not warn for duplicate assigmentens in "used" files

### PR DESCRIPTION
**Pull request by MichaelPFrey**
change on the parser: distinguish between mainModule and included modules

### CODE REVIEW

**Comment by kintel** [comment]
Two questions:
* Are we covering enough cases where such warnings should be issued? (See discussion in #2513)
* This manages a subset of possible overwrites; could we clarify in a code comment what we're actually trying to achieve?

File: src/parser.y [hunk 1]
```diff
@@ -661,10 +662,12 @@ void yyerror (char const *s)
          (*sourcefile()) % lexerget_lineno() % s);
 }
 
-bool parse(FileModule *&module, const char *text, const std::string &filename, int debug)
+bool parse(FileModule *&module, const char *text, const std::string &filename, int debug, bool mainModule)
 {
   fs::path path = fs::absolute(fs::path(filename));
   
+  bMainModule = mainModule;
```
Comment (kintel) [inline src/parser.y]: Would it be cleaner to instead pass the `assignmentWarning` flag as an argument, to have one less piece of logic to manage?

File: src/parser.y [hunk 2]
```diff
@@ -215,13 +216,13 @@ assignment:
                         auto prevFile = assignment.location().fileName();
                         auto currFile = LOC(@$).fileName();
                         
-                        if(assignmentWarning && prevFile==RootFile && currFile == RootFile){
+                        if(bMainModule && assignmentWarning && prevFile==RootFile && currFile == RootFile){
```
Comment (kintel) [inline src/parser.y]: `assignmentWarning && prevFile==RootFile` is shared between the if and else branches. Pulling this out could make the code a bit more readable.

**Comment by MichaelPFrey** [comment]
I see that as somewhat of a hotfix for the confusing duplicate assignment warning, when a variable is overwritten within a "use<>", not as a final solution covering all bases.

#2492 it self was as much a feature as it is testing the water.  #2513 shows support for a verbose warning system. Knowing that, I am willing to go the extra mile and try to cover more cases of duplicate assignments.

**Comment by MichaelPFrey** [comment]
replaced by #2516 

Keeped for reference, as #2516 is a more complicated patch, based on this one.

**Pull request closed without merging**